crates/ibc/src/core/ics02_client/msgs/update_client.rs

```diff
@@ -15,13 +15,14 @@ pub const TYPE_URL: &str = "/ibc.core.client.v1.MsgUpdateClient";
 
 /// A type of message that triggers the update of an on-chain (IBC) client with new headers.
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct MsgUpdateClient {
+pub(crate) struct MsgUpdateClient {
     pub client_id: ClientId,
     pub header: Any,
     pub signer: Signer,
 }
 
 impl MsgUpdateClient {
+    #[allow(dead_code)]
```

Comment:
We should just remove the constructor if it's unused. Same comment applies to all uses on `#[allow(dead_code)]`